### PR TITLE
Delete users

### PR DESCRIPTION
**Pull request by connor-hawley**
Added a button to the edit roster view to delete users. Also added a confirmation modal with extra confirmation before the user is able to delete user. Extended the API to allow the new delete user endpoint to be called.

### CODE REVIEW

File: src/api/UsersAPI.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import APIClient from "./APIClient";
 import Cookies from "universal-cookie";
+import { tsThisType } from "@babel/types";
```
Comment (dbecker1) [inline src/api/UsersAPI.js:3]: What is this for?

Comment (connor-hawley) [reply src/api/UsersAPI.js:3]: I have no idea. Probably some vscode autocomplete autoimport BS. I'll remove it real quick

**Comment by connor-hawley** [comment]
While we're on the subject, do you want prs to be for the dev branch for the frontend too in the future?

**Comment by dbecker1** [comment]
Yeah my preferred flow would be either make PRs into dev or push directly to dev and then make the official reviewed PR be the one from dev to master. That way I can pull up the dev site and verify functionality

**Pull request merged**